### PR TITLE
Add Multiple Sentiment/Hate Speech Classification Datasets

### PR DESCRIPTION
**Pull request by sted97**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Increase multilingual coverage of the MMTEB benchmark by adding 3 new datasets about Sentiment/Hate Speech Classification spanning 8 languages (3 new).

Specifically, the following 3 datasets have been added to this PR:
1. [German Politicians Twitter Sentiment](https://huggingface.co/datasets/Alienmaster/german_politicians_twitter_sentiment)
2. [Hate Speech Portuguese](https://huggingface.co/datasets/hate_speech_portuguese)
3. [Tweet Sentiment Multilingual](https://huggingface.co/datasets/cardiffnlp/tweet_sentiment_multilingual)

I have evaluated the models on the 3 datasets as specified in the checklist, and added the results in the PR.

Concerning the points, I computed them as follows: 6 for the German dataset (2 + 4 for the new language), 6 for the Portuguese dataset (2+4), and 20 (2*7 + 6) for the multilingual dataset with 8 languages (1 new). I'm not sure about the calculations, please correct me if I'm wrong.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
> Concerning the points, I computed them as follows: 6 for the German dataset (2 + 4 for the new language), 6 for the Portuguese dataset (2+4), and 20 (2*7 + 6) for the multilingual dataset with 8 languages (1 new). I'm not sure about the calculations, please correct me if I'm wrong.

I believe german, portugese and  is already covered by clf for the multilingual dataset you only get 2 points for the dataet and 4 pr. new language. I believe this makes the points 2+2+(2+4) (what is the new language in the dataset)

File: docs/mmteb/points/598.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
+{"GitHub": "sted97", "New dataset": 32}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/598.jsonl:1]: update (see comment below and add points)

File: mteb/tasks/Classification/multilingual/TweetSentimentClassification.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                publisher = "European Language Resources Association",
+                url = "https://aclanthology.org/2022.lrec-1.27",
+                pages = "258--266",
+                abstract = "Language models are ubiquitous in current NLP, and their multilingual capacity has recently attracted considerable attention. However, current analyses have almost exclusively focused on (multilingual variants of) standard benchmarks, and have relied on clean pre-training and task-specific corpora as multilingual signals. In this paper, we introduce XLM-T, a model to train and evaluate multilingual language models in Twitter. In this paper we provide: (1) a new strong multilingual baseline consisting of an XLM-R (Conneau et al. 2020) model pre-trained on millions of tweets in over thirty languages, alongside starter code to subsequently fine-tune on a target task; and (2) a set of unified sentiment analysis Twitter datasets in eight different languages and a XLM-T model trained on this dataset.",
+            }
+        """,
+        n_samples={"validation": 2590, "test": 6960},
+        avg_character_length={"validation": 86.27, "test": 83.51},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/TweetSentimentClassification.py:59]:
```suggestion
        n_samples={"test": 6960},
        avg_character_length={"test": 83.51},
```
we are only using the test

File: mteb/tasks/Classification/multilingual/TweetSentimentClassification.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            "revision": "14b13edfbc4046892f6011d114c29c0f83170589",
+        },
+        description="A multilingual Sentiment Analysis dataset consisting of tweets in 8 different languages.",
+        reference="https://aclanthology.org/2022.lrec-1.27",
+        category="s2s",
+        type="Classification",
+        eval_splits=["test"],
+        eval_langs=_LANGUAGES,
+        main_score="accuracy",
+        date=("2022-03-18", "2022-03-20"),
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/TweetSentimentClassification.py:33]: the date of when the text data was written created (can be your best guess). Are they only really sampled from 2 days?

File: mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                month = aug,
+                year = "2019",
+                address = "Florence, Italy",
+                publisher = "Association for Computational Linguistics",
+                url = "https://aclanthology.org/W19-3510",
+                doi = "10.18653/v1/W19-3510",
+                pages = "94--104",
+            }
+        """,
+        n_samples={"train": 5670},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py:53]: Please downsample using the suggested function in the checklist

File: mteb/tasks/Classification/multilingual/TweetSentimentClassification.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/multilingual/TweetSentimentClassification.py:59]: please downsample here as well

File: mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        description="HateSpeechPortugueseClassification is a dataset of Portuguese tweets categorized with their sentiment (2 classes).",
+        reference="https://aclanthology.org/W19-3510",
+        dataset={
+            "path": "hate_speech_portuguese",
+            "revision": "b0f431acbf8d3865cb7c7b3effb2a9771a618ebc",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["prt-Latn"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py:20]: I don't believe prt is portugese.

File: mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        main_score="accuracy",
+        date=("2017-03-08", "2017-03-09"),
+        form=["written"],
+        domains=["Social"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license="Not specified",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py:26]: Is this really the case?

**Comment by sted97** [comment]
> > Concerning the points, I computed them as follows: 6 for the German dataset (2 + 4 for the new language), 6 for the Portuguese dataset (2+4), and 20 (2*7 + 6) for the multilingual dataset with 8 languages (1 new). I'm not sure about the calculations, please correct me if I'm wrong.
> 
> I believe german, portugese and is already covered by clf for the multilingual dataset you only get 2 points for the dataet and 4 pr. new language. I believe this makes the points 2+2+(2+4) (what is the new language in the dataset)

Oh, I see. Thanks a lot for the clarification, Kenneth! Then it is 2+2 for the German and Portuguese datasets; I checked if these languages had specific subfolders in the Classification folder without checking the languages covered by the multilingual datasets. 

The new language for the multilingual dataset was Spanish, but I just checked and it is included in other multilingual datasets. I made the multiplication 2*7 only because I saw it in PR #512, but if it should not be multiplied, the total is 2+2+2=6.

File: mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py (hunk 6 above)
Comment (sted97) [reply mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py]: Yes, unfortunately the license is not specified either in the dataset tab or in the GitHub repo.

File: mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py (hunk 5 above)
Comment (sted97) [reply mteb/tasks/Classification/prt/HateSpeechPortugueseClassification.py]: Fixed: changed to "por"

File: mteb/tasks/Classification/multilingual/TweetSentimentClassification.py (hunk 3 above)
Comment (sted97) [reply mteb/tasks/Classification/multilingual/TweetSentimentClassification.py]: In the paper the authors say "We used the Twitter API to retrieve 198M tweets posted between May’18 and March’20 [...]". I had read it wrong. Now it is fixed

**Comment by KennethEnevoldsen** [comment]
Thanks, I have added a comment on #512 (generally it has not been the case though).

File: docs/mmteb/points/598.jsonl [hunk 7]
```diff
@@ -0,0 +1 @@
+{"GitHub": "sted97", "New dataset": 20}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/598.jsonl:1]: update and add review scores

**Comment by sted97** [comment]
> Thanks, I have added a comment on #512 (generally it has not been the case though).

Perfect! :D

File: mteb/tasks/Classification/multilingual/TweetSentimentClassification.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                month = jun,
+                year = "2022",
+                address = "Marseille, France",
+                publisher = "European Language Resources Association",
+                url = "https://aclanthology.org/2022.lrec-1.27",
+                pages = "258--266",
+                abstract = "Language models are ubiquitous in current NLP, and their multilingual capacity has recently attracted considerable attention. However, current analyses have almost exclusively focused on (multilingual variants of) standard benchmarks, and have relied on clean pre-training and task-specific corpora as multilingual signals. In this paper, we introduce XLM-T, a model to train and evaluate multilingual language models in Twitter. In this paper we provide: (1) a new strong multilingual baseline consisting of an XLM-R (Conneau et al. 2020) model pre-trained on millions of tweets in over thirty languages, alongside starter code to subsequently fine-tune on a target task; and (2) a set of unified sentiment analysis Twitter datasets in eight different languages and a XLM-T model trained on this dataset.",
+            }
+        """,
+        n_samples={"test": 6960},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/TweetSentimentClassification.py:58]:
```suggestion
        n_samples={"test": 2048},
```

File: mteb/tasks/Classification/por/HateSpeechPortugueseClassification.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                month = aug,
+                year = "2019",
+                address = "Florence, Italy",
+                publisher = "Association for Computational Linguistics",
+                url = "https://aclanthology.org/W19-3510",
+                doi = "10.18653/v1/W19-3510",
+                pages = "94--104",
+            }
+        """,
+        n_samples={"train": 5670},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/por/HateSpeechPortugueseClassification.py:53]:
```suggestion
        n_samples={"train": 2048},
```

**Review by KennethEnevoldsen — COMMENTED** [review]
A few minor fixes

**Pull request merged**